src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -1166,7 +1170,8 @@ def build_udtf(
     comment: Optional[str] = None,
     statement_params: Optional[Dict[str, str]] = None,
     is_permanent: bool = False,
-    session=None,
+    session: "snowflake.snowpark.session.Session" = None,
+    udtf_name: Optional[Union[str, Iterable[str]]] = None,
```

Comment:
can we rename to not clash with `name`? Simply doing `_registered_object_name` should do.